query/definition_resolution.rs

```diff
@@ -0,0 +1,474 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+// TODO: not sure if these go elsewhere, or are useful in lower level packges outside //query
```

Comment:
This is a good point - maybe it should move elsewhere later! Not right now though I think!